x-pack/platform/plugins/shared/alerting_v2/server/plugin.ts

```diff
@@ -31,21 +33,38 @@ export class AlertingPlugin
     >
 {
   private readonly logger: Logger;
+  private readonly serviceManager: ServiceManager;
 
   constructor(initializerContext: PluginInitializerContext) {
-    this.logger = initializerContext.logger.get();
-    void this.logger;
+    this.logger = initializerContext.logger.get('alerting_v2');
+    this.serviceManager = new ServiceManager();
   }
 
   public setup(core: CoreSetup, plugins: AlertingServerSetupDependencies) {
     registerFeaturePrivileges(plugins.features);
+
+    registerDirectorTask(
+      plugins.taskManager,
+      () => this.serviceManager.getDirectorService(),
```

Comment:
Can the task run before the plugin executes the `start` method, where the initialization happens?